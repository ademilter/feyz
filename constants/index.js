export const PATHS = [
  { slug: 'tasarim', title: 'Tasarım' },
  { slug: 'frontend', title: 'Frontend' },
  { slug: 'yazilim', title: 'Yazılım' },
  { slug: 'turkce-kaynak', title: 'Türkçe Kaynak' },
  { slug: 'video', title: 'Video' },
  { slug: 'podcast', title: 'Podcast' },
  { slug: 'yazi', title: 'Blog Yazısı' },
  { slug: 'arac', title: 'Araç' },
  { slug: 'etkinlik', title: 'Etkinlik' },
  { slug: 'tweet', title: 'Tweet' },
  { slug: 'alinti', title: 'Alıntı' }
]
