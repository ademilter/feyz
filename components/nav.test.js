import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import Nav from './nav'
import { PATHS } from '../constants'

describe('Nav', () => {
  test('should have children.', () => {
    const randomElement = PATHS[Math.floor(Math.random() * PATHS.length)]
    const { asFragment } = render(<Nav />)
    const element = screen.getByText(new RegExp(randomElement.title, 'i'))

    expect(asFragment()).toMatchSnapshot()
    expect(screen.queryByTestId('nav').childElementCount).toBe(PATHS.length + 1)
    expect(element).not.toBeNull()
  })
})
