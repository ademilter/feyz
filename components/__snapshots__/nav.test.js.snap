// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should have children. 1`] = `
<DocumentFragment>
  <div
    class=""
    data-testid="nav"
  >
    <a
      href="/tag/tasarim"
    >
      Tasarım
    </a>
    <a
      href="/tag/frontend"
    >
      Frontend
    </a>
    <a
      href="/tag/yazilim"
    >
      Yazılım
    </a>
    <a
      href="/tag/turkce-kaynak"
    >
      Türkçe Kaynak
    </a>
    <a
      href="/tag/video"
    >
      Video
    </a>
    <a
      href="/tag/podcast"
    >
      Podcast
    </a>
    <a
      href="/tag/yazi"
    >
      Blog Yazısı
    </a>
    <a
      href="/tag/arac"
    >
      Araç
    </a>
    <a
      href="/tag/etkinlik"
    >
      Etkinlik
    </a>
    <a
      href="/tag/tweet"
    >
      Tweet
    </a>
    <a
      href="/tag/alinti"
    >
      Alıntı
    </a>
    <a
      class=""
      href="https://www.patreon.com/ademilter"
      rel="noopener noreferrer"
      target="_blank"
    >
      Destek Ol
    </a>
  </div>
</DocumentFragment>
`;
